src/sentry/ratelimits/cardinality.py

```diff
@@ -408,7 +408,8 @@ def run_check_within_quotas(
         self, unit_keys_to_get: Sequence[str], set_keys_to_count: Sequence[str]
     ) -> Tuple[Mapping[str, int], Mapping[str, int]]:
         with self.client.pipeline(transaction=False) as pipeline:
-            pipeline.mget(unit_keys_to_get)
+            for key in unit_keys_to_get:
```

Comment:
I don't think this should pass tests, `unit_keys` needs to be adapted to call `next(results)` exactly `len(unit_keys_to_get)` times instead of calling it once and assuming the return value is the list of unit keys

if it does pass tests we have large coverage problems